src/routes/emailAuthData.ts

```diff
@@ -0,0 +1,44 @@
+import { celebrate, Joi, Segments } from "celebrate";
+import express from "express";
+import log from "loglevel";
+
+import redis from "../database/redis";
+import { getError, REDIS_NAME_SPACE } from "../utils";
+
+const router = express.Router();
+const REDIS_TIMEOUT = 300; // 5m
+
+router.post(
+  "/updateAuthData",
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      encAuthData: Joi.string().required(),
+      instancePubKeyX: Joi.string().max(66).required(),
```

Comment:
we should use length instead of max wherever possible.

Reply:
Cannot always do this because signatures can have 0 at the start which when converted to hex are not included.
hence, we must always use max